### PR TITLE
Show all errors, even if there are multiple

### PR DESCRIPTION
**Pull request by jez**
For split fields if there are ever multiple error messages from the
individual fields, we want to remember them. This way, if we ever
overwrite one with another, we can come back and start showing the old
one again once they've fixed the new one.

Previously, once you fixed the new one it would show you no message, as
if the form were valid.

r? @atty-stripe

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -34,13 +34,28 @@ function registerElements(elements, exampleName) {
...
     element.on('change', function(event) {
       if (event.error) {
         error.classList.add('visible');
+        savedErrors[idx] = event.error.message;
         errorMessage.innerText = event.error.message;
       } else {
-        error.classList.remove('visible');
+        savedErrors[idx] = null;
+
+        var maybeError = Object.keys(savedErrors).reduce(function(acc, key) {
```
Comment (atty-stripe) [inline js/index.js]: s/maybeError/nextError/?

Also would be useful to add a one-line comment explaining what this reduce is doing. Maybe also rename acc to accumulator.

File: js/index.js [hunk 2]
```diff
@@ -34,13 +34,28 @@ function registerElements(elements, exampleName) {
   }
 
   // Listen for errors from each Element, and show error messages in the UI.
-  elements.forEach(function(element) {
+  var savedErrors = {};
```
Comment (atty-stripe) [inline js/index.js]: We could drop this I think

**Comment by jez** [comment]
ptal @atty-stripe

**Comment by atty-stripe** [comment]
LGTM

**Pull request merged**